### PR TITLE
Fix #1672 BackgroundSwitcher review

### LINKED ISSUE #1672: BackgroundSwitcher review
Background switcher floating on the map, with the current background preview showing. When clicked the tool should display the alternative backgrounds. Should be perfect if it is somehow merged with the overview map.

### PR DESCRIPTION
**Pull request by allyoucanmap**

### CODE REVIEW

File: web/client/plugins/BackgroundSelector.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  *       "side": 78,
+  *       "sidePreview": 104,
+  *       "frame": 3,
+  *       "margin": 5
+  *     }
+  *   }
+  * }
+  */
+
+const BackgroundSelectorPlugin = connect((state) => ({
```
Comment (offtherailz) [inline web/client/plugins/BackgroundSelector.jsx]: use reselect

File: web/client/components/background/BackgroundSelector.jsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            size: {width: 0, height: 0},
+            desktop: {},
+            mobile: {},
+            onToggle: () => {},
+            propertiesChangeHandler: () => {},
+            setControlProperty: () => {}
+        };
+    },
+    componentWillUnmount() {
+        this.props.setControlProperty('backgroundSelector', 'currentLayer', {});
```
Comment (offtherailz) [inline web/client/components/background/BackgroundSelector.jsx]: use a separate reducer or bind this method to background selector from the plugin. I don't want to make components aware of the setControlProperty method if it is possible.

File: web/client/components/background/PaginationButton.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    getDefaultProps() {
+        return {
+            side: 0,
+            direction: true,
+            vertical: false,
+            onClick: () => {}
+        };
+    },
+    render() {
+        let glyph = this.props.direction ? 'chevron-right' : 'chevron-left';
```
Comment (mbarto) [inline web/client/components/background/PaginationButton.jsx]: const glyph = glyphs[this.props.vertical][this.props.direction];

File: web/client/components/background/PreviewIcon.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            layer: {},
+            currentLayer: {},
+            onToggle: () => {},
+            onClose: () => {},
+            setLayer: () => {}
+        };
+    },
+    render() {
+        const containerClass = this.props.vertical ? 'background-preview-icon-container-vertical' : 'background-preview-icon-container-horizontal';
+        let type = this.props.layer.visibility ? containerClass + ' bg-primary' : containerClass + ' bg-body';
```
Comment (mbarto) [inline web/client/components/background/PreviewIcon.jsx]: Use const, let is not needed. You can do everything without calculating type twice

File: web/client/plugins/DrawerMenu.jsx [hunk 5]
```diff
@@ -78,9 +79,13 @@ const DrawerMenu = React.createClass({
             menuOptions: {},
             singleSection: true,
             buttonClassName: "square-button",
-            disabled: false
+            disabled: false,
+            resetWidth: () => {}
         };
     },
+    componentWillUnmount() {
+        this.props.resetWidth();
```
Comment (mbarto) [inline web/client/plugins/DrawerMenu.jsx]: I don't think this is needed. If the control is not enabled, width is 0, we don't need an additional property

**Pull request merged**